### PR TITLE
fix: bump to core 3

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
bumps to `sfdx-core@3`
removes unused files
fixes jsdocs
refactors a few unnecessary variables

### What issues does this PR fix or reference?
@W-11092409@

### CODE REVIEW

File: src/appInsights.ts [hunk 1]
```diff
@@ -36,8 +36,6 @@ export interface TelemetryOptions {
   waitForConnection?: boolean;
 }
 
-Messages.importMessagesDirectory(__dirname);
```
Comment (WillieRuemmele) [inline src/appInsights.ts:39]: the messages were never being used

File: package.json [hunk 2]
```diff
@@ -33,33 +33,35 @@
...
+    "@salesforce/core": "^3.15.5",
+    "@salesforce/ts-types": "^1.5.20",
     "applicationinsights": "^1.4.0",
     "axios": "^0.27.2"
   },
   "devDependencies": {
-    "@salesforce/dev-config": "^2.0.0",
-    "@salesforce/dev-scripts": "^0.9.18",
+    "@salesforce/dev-config": "^3.0.1",
+    "@salesforce/dev-scripts": "^2.0.2",
```
Comment (cristiand391) [inline package.json:43]: The `prepare` script should be set to `sf-install` to make dev-scripts work (it's missing the husky folder).

**Comment by cristiand391** [comment]
@WillieRuemmele do you want to release this as minor v3 or as telemetry v4? Make sure to manually bump the version if v4.

**Review by cristiand391 — APPROVED** [review]

**Comment by cristiand391** [comment]
**QA notes**:

Checkout https://github.com/salesforcecli/plugin-telemetry/pull/282 and yarn link this PR:

✅ lib respects `SFDX_DISABLE_INSIGHTS=true`
❌ lib respects `disableTelemetry=true` setting
❌ when telemetry is disabled, `./bin/run telemetry | sfdx telemetry` prints the ID as undefined

Screen Shot 2022-05-05 at 12 57 51

With telemetry enabled:

✅ lib respects `SFDX_TELEMETRY_DEBUG` and can run the uploader manually, also looked at the log and all the event props are properly populated:
Screen Shot 2022-05-05 at 13 07 47

✅ lib respects `SFDX_TELEMETRY_PATH`:
Screen Shot 2022-05-05 at 13 11 04

✅ lib respects `SFDX_TELEMETRY_DISABLE_ACKNOWLEDGEMENT` when the ack file doesn't exists (first run):
Screen Shot 2022-05-05 at 13 15 04

**Comment by cristiand391** [comment]
**QA notes**:

With oclif/core PR linked:

✅ plugin-telemetry respects `disableTelemetry` setting:

`disableTelemetry=true`

Screen Shot 2022-05-06 at 10 59 58

`disableTelemetry=false`

Screen Shot 2022-05-06 at 11 01 54

**Comment by cristiand391** [comment]
**QA notes:**

With oclif/core PR linked

✅ plugin-telemetry respects `disableTelemetry` setting:
Screen Shot 2022-05-06 at 11 09 25

**Pull request merged**